draft-parecki-oauth-first-party-native-apps.md

```diff
@@ -370,6 +370,12 @@ The device session is completely opaque to the client, and as such the AS MUST a
 
 The client MUST include the device session in future requests to the authorization challenge endpoint for the particular authorization request.
 
+## Authorization Challenge Grant - Client Metadata
+
+This specification introduces a new OAuth 'grant_type' named 'authz_challenge' to the client metadata{{RFC7591}}. This grant enables the authorization server to identify if a client is authorized to use the authorization challenge flow. In order to reduce the risk of unauthorized client use, and, attackers from abusing the authorization challenge endpoint, this grant type MUST only be issued to first party app clients.
```

Comment:
One of the goals of the effort is to explicitly NOT introduce a new grant_type. The goal is to allow all the standard existing OAuth2 specifications to work with this new endpoint including using the standards 'code' grant_type. Please file an issue so we can discuss more in depth.

Reply:
Thanks. I wasn't aware that was one of the goals. I'll open an issue to discuss the intent further.